src/plugins/explore/public/components/tabs/action_bar/results_action_bar/results_action_bar.tsx

```diff
@@ -77,30 +77,35 @@ export const DiscoverResultsActionBar = ({
               })}
             </EuiButtonEmpty>
           </EuiFlexItem> */}
-          {dataset && rows?.length ? (
-            <>
-              {shouldShowExportButton && (
-                <EuiFlexItem
-                  grow={false}
-                  className="explore-results-action-bar__explore-download-csv-flex-item"
-                >
-                  <DiscoverDownloadCsv indexPattern={dataset as any} rows={rows} hits={hits} />
-                </EuiFlexItem>
-              )}
-
-              {showTabSpecificSettings && (
-                <EuiFlexItem grow={false}>
-                  <PatternsSettingsPopoverButton />
-                </EuiFlexItem>
-              )}
-
-              {shouldShowAddToDashboardButton && (
-                <EuiFlexItem grow={false}>
-                  <SaveAndAddButtonWithModal dataset={dataset} />
-                </EuiFlexItem>
-              )}
-            </>
-          ) : null}
+          <EuiFlexItem grow={false}>
+            {dataset && rows?.length ? (
+              <EuiFlexGroup
+                alignItems="center"
+                direction="row"
+                gutterSize="none"
+                justifyContent="flexStart"
+              >
+                {showTabSpecificSettings && (
+                  <EuiFlexItem grow={false}>
+                    <PatternsSettingsPopoverButton />
+                  </EuiFlexItem>
+                )}
+                {shouldShowExportButton && (
+                  <EuiFlexItem
+                    grow={false}
+                    className="explore-results-action-bar__explore-download-csv-flex-item"
+                  >
+                    <DiscoverDownloadCsv indexPattern={dataset as any} rows={rows} hits={hits} />
```

Comment:
i don thtink it needs any for dataset. it should be the same

Reply:
gotcha. ignore the following if you don't have any other changes.

but if you need to push more changes do you mind cleaning this part up 🥺

Reply:
cleaning it up! 😁